### PR TITLE
Decouple component context tests from global resolver.

### PR DESCRIPTION
**Pull request by patricksrobertson**
This extracts the Application Lifecycle - Component Context tests
from the component registration tests and then rewrites them to use
the TestModuleResolver.

side note: I have no idea what I'm doing here.

### CODE REVIEW

File: packages/ember/tests/component_context_test.js [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+
+    this.add('controller:application', Controller.extend({
+      'text': 'outer'
+    }));
+    this.addComponent('my-component', { ComponentClass: Component.extend({
+      text: 'inner'
+    }), template: '{{text}}-{{yield}}' });
+
+    this.visit('/').then(() => {
+      assert.equal(jQuery('#wrapper').text(), 'inner-outer', 'The component is composed correctly');
```
Comment (mixonic) [inline packages/ember/tests/component_context_test.js]: `jQuery` here can be replaced with: `this.$('#wrapper')`. The latter will scope the selector (and avoid importing `jQuery`). Also elsewhere through this patch.

File: packages/ember/tests/component_context_test.js [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+moduleFor('Application Lifecycle - Component Context', class extends ApplicationTestCase {
+  ['@test Components with a block should have the proper content when a template is provided'](assert) {
+    this.addTemplate('application', '<div id=\'wrapper\'>{{#my-component}}{{text}}{{/my-component}}</div>');
+
+    this.add('controller:application', Controller.extend({
+      'text': 'outer'
+    }));
+    this.addComponent('my-component', { ComponentClass: Component.extend({
+      text: 'inner'
+    }), template: '{{text}}-{{yield}}' });
```
Comment (mixonic) [inline packages/ember/tests/component_context_test.js]: Perhaps can this be formatted to be more legible?

```js
this.addComponent('my-component', {
  ComponentClass: Component.extend({
    text: 'inner'
  }),
  template: `{{text}}-{{yield}}`
});
```

File: packages/ember/tests/component_context_test.js [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+      }
+    }) });
+
+    this.visit('/').then(() => {
+      assert.equal(jQuery('#wrapper').text(), 'Some text inserted by jQuery', 'The component is composed correctly');
+    });
+  }
+
+  ['@test Components trigger actions in the parents context when called from within a block'](assert) {
+    this.addTemplate('application', '<div id=\'wrapper\'>{{#my-component}}<a href=\'#\' id=\'fizzbuzz\' {{action \'fizzbuzz\'}}>Fizzbuzz</a>{{/my-component}}</div>');
```
Comment (mixonic) [inline packages/ember/tests/component_context_test.js]: Wrapping the templates in backticks would avoid all the quote escaping here. In addition backticks would permit multiple lines:

```js
this.addTemplate('application', `
  <div id="wrapper">
    {{#my-component}}
      <a href="#" id="fizzbuzz" {{action 'fizzbuzz'}}>
        Fizzbuzz
      </a>
    {{/my-component}}
  </div>
`);
```

Much easier to grok.

File: packages/ember/tests/component_context_test.js [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+    this.addComponent('my-component', { ComponentClass: Component.extend({
+      actions: {
+        fizzbuzz() {
+          ok(true, 'action triggered on component');
+        }
+      }
+    }), template: '<a href=\'#\' id=\'fizzbuzz\' {{action \'fizzbuzz\'}}>Fizzbuzz</a>' });
+
+    this.visit('/').then(() => {
+      run(() => jQuery('#fizzbuzz', '#wrapper').click());
```
Comment (mixonic) [inline packages/ember/tests/component_context_test.js]: runloop should not be needed here, or around any other `click` statements. `this.$('#fizzbuzz').click()` would be fine.

**Review by mixonic — CHANGES_REQUESTED** [review]
@patricksrobertson this is a solid start thank you! I made a few notes of how to clean it up to be ready for a merge.

This commits should be squashed as well.

**Comment by patricksrobertson** [comment]
@mixonic should be all set. Since we went with backticks for the templates, I opted to go:
```js
let text = this.$('#wrapper').text().trim()
assert.equal(text, ...)
```
to not change the assertions and to save line width.

**Review by mixonic — APPROVED** [review]

**Comment by mixonic** [comment]
Excellent! Thank you @patricksrobertson :-)

**Pull request merged**